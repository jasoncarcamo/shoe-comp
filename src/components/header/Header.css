header{
    width: 100%;
    margin: 0;
}

#header-nav{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    text-align: center;
}

#header-nav li{
    list-style: none;
}

#header-nav li *{
    text-decoration: none;
}

.logging-container{
    display: flex;
    width: 10em;
    justify-content: space-evenly;
    margin-left: auto;
    text-align: right;
}

.logging-link{
    padding-right: 1em;
    color: black;
}

#header-nav > h2 *{
    width: 50%;
    text-align: center;
    margin: 0 auto .5em auto;
    font-size: 2.3em;
    color: purple;
    text-decoration: none;
}

/*Checkout button section*/
#cart-container{
    position: fixed;
    top: 5em;
    right: 0;
    margin-left: auto;
    width: 5em;
    padding-right: 2em;
}

#cart-amount{
    color: red;
    position: relative;
    top: .2em;
    padding: 0 1em;
    margin: 0 auto;
    font-size: 1.8em;
}

#cart-icon{
    display: block;
    font-size: 1.6em;
    padding: 0;
    margin: 0 auto;
}

#checkout-container{
    display: none;
}

#checkout-container > div{
    width: 1em;
    height: 1em;
    transform: rotate(45deg);
    margin: 0 auto;
    background-color: black;
    z-index: -1;
}

#checkout-container > span {
    display: block;
    position: relative;
    right: 1.2em;
    width: 7em;
    height: 2.5em;
    padding-top: 1em;
    margin-top: -.5em;
    background-color: black;
    z-index: 4;
}

.to-checkout{
    display: block;
    width: 5em;
    height: 1.5em;
    margin: 0 auto 2em auto;
    font-size: 1em;
    color: white;
    text-decoration: none;
    border-radius: 1em;
    border: .1em solid white;
}


/*Nav links section*/
#nav-links{
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    width: 50%;
    text-align: center;
    margin: 0 auto;
    padding: 0;
}

#nav-links *{
    color: black;
}

@media screen and (max-width: 770px){

    #nav-links{
        display: none;
    }

    .showUl{
        display: flex !important;
    }

    #header-nav h2{
        margin-top: 1.5em;
    }

    .logging-container{
        position: absolute;
        right: 1em;
        z-index: 3;
    }

    #nav-links{
        position: fixed;
        z-index: 2;
        min-height: 100%;
        width: 100%;
        background-color: white;
    }

    #cart-container{
        top: .1em;
        right: 1em;
    }

    /* Menu icon - mobile view */

    #menu-icon{
        position: fixed;
        top: 2.5em;
        right: 1.5em;
        width: 1.7em;
        z-index: 3;
    }

    #menu-icon div{
        width: 1.7em;
        margin: .2em 0;
        height: .3em;
        background-color: black;
    }
}