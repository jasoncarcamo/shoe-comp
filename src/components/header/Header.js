import React from 'react';
import {Link, NavLink} from 'react-router-dom';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faShoppingCart} from "@fortawesome/free-solid-svg-icons";
import './Header.css';
import TokenService from '../../services/TokenService';
import CartContext from "../CartContext/CartContext";

export default class Header extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            items: [],
        }
    };
    
    componentDidMount(){
        
    }

    componentWillUnmount(){
        
    }

    static contextType = CartContext;

    amountOfItems = ()=>{
        let amount = this.context.items.length;

        if(amount === 0){
            amount = null;
        };
        
        return amount;
    }

    renderCart = () => {
            return (
                <div id="cart-container">
                    <span id="cart-amount">{this.amountOfItems()}</span>

                    <FontAwesomeIcon className="fas-10x" id="cart-icon" icon={faShoppingCart} onClick={this.toggleCheckout}></FontAwesomeIcon>

                    <div id="checkout-container">
                        <div></div>
                        <span>
                            <Link className="to-checkout" to="/checkout" onClick={this.toggleCheckout}>Check out</Link>
                        </span>
                    </div>
                </div>
            )
    }

    toggleCheckout = ()=>{
        const checkout = document.getElementById("checkout-container");

        if(checkout.style.display !== "block"){
            checkout.style.display = "block";
        } else{
            checkout.style.display = "none";
        }
    }

    handleLogOut = ()=>{
        TokenService.clearAuthToken();
        this.props.history.push("/");
    }

    render(){

        return (
            <header>
                
                <nav id="header-nav">
                    <ul className="logging-container">
                        {TokenService.hasAuthToken() ? <li><Link to="/" className="logging-link" onClick={this.handleLogOut}>Log Out</Link></li> : <>  
                    <li><Link to="/register" className="logging-link">Sign Up</Link></li>
                    <li><Link to="/login" className="logging-link">Log In</Link></li>
                </>}
                    </ul>
                    
                    <h2>Icon</h2>

                    {TokenService.hasAuthToken() ? this.renderCart() : null}

                    <ul id="nav-links">

                        <li><NavLink exact to="/" activeStyle={{fontSize: "1.2em", fontWeight: 700}}>Home</NavLink></li>

                        <li>
                            <NavLink to="/shop" activeStyle={{fontSize: "1.2em", fontWeight: 700}}>Shop</NavLink>
                        </li>

                        {TokenService.hasAuthToken() ? <li>
                            <NavLink to="/user" activeStyle={{fontSize: "1.2em", fontWeight: 700}}>Profile</NavLink>
                        </li> : ''}
                        <li><NavLink to="/about" activeStyle={{fontSize: "1.2em", fontWeight: 700}}>About Us</NavLink></li>
                    </ul>
                </nav>
            </header>
        )
    }
}