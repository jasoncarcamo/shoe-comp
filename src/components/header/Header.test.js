import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import Header from "./Header";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter()});


describe("<Header/>", ()=>{

    it("Renders without crashing", ()=>{
        const div = document.createElement("div");

        ReactDOM.render(<BrowserRouter><Header/></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Creates a snapshot", ()=>{
        const tree = renderer.create(<BrowserRouter><Header/></BrowserRouter>).toJSON()

        expect(tree).toMatchSnapshot();
    });

    
});