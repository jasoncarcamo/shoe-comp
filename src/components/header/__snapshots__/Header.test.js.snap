// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Creates a snapshot 1`] = `
<header>
  <nav
    id="header-nav"
  >
    <ul
      className="logging-container"
    >
      <li>
        <a
          className="logging-link"
          href="/register"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </li>
      <li>
        <a
          className="logging-link"
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
      </li>
    </ul>
    <h2>
      Icon
    </h2>
    <ul
      id="nav-links"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "fontSize": "1.2em",
              "fontWeight": 700,
            }
          }
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/shop"
          onClick={[Function]}
        >
          Shop
        </a>
      </li>
      
      <li>
        <a
          aria-current={null}
          href="/about"
          onClick={[Function]}
        >
          About Us
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
